showcase/tests/integration/components/hds/stepper/nav-panel-test.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+// NOTICE
+// Because of how the `panel` subcomponent is built,
+// it's practically impossible to test in isolation, so in our tests
+// in this file it will be wrapped inside its parent comoponent.
+
+module('Integration | Component | hds/stepper/nav/panel', function (hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function () {
+    this.set('createNavPanel', async (args = {}) => {
+      this.currentStep = args.currentStep ?? undefined;
+      return await render(hbs`
+          <Hds::Stepper::Nav @currentStep={{this.currentStep}} as |S|>
+            <S.Step></S.Step>
+            <S.Panel>
+              <div id="test-panel-content">Test</div>
+            </S.Panel>
+          </Hds::Stepper::Nav>
+        `);
+    });
+  });
+
+  // CLASSES
+
+  test('it should render the component with a CSS class that matches the component name', async function (assert) {
+    await render(hbs`<Hds::Stepper::Nav::Panel data-test="panel-1" />`);
+    assert.dom('[data-test="panel-1"]').hasClass('hds-stepper-nav__panel');
+  });
+
+  // VISIBILITY
+
+  test('it sets the panel content to not visible when the @currentStep argument does not match the panel index in the @panelIds argument', async function (assert) {
+    await this.createNavPanel({ currentStep: 1 });
+    assert.dom('.hds-stepper-nav__panel').hasAttribute('hidden');
+    assert.dom('#test-panel-content').doesNotExist();
+  });
+
+  test('it sets the panel content to visible when the @currentStep argument matches the panel index in the @panelIds argument', async function (assert) {
+    await this.createNavPanel({ currentStep: 0 });
+    assert.dom('.hds-stepper-nav__panel').hasNoAttribute('hidden');
+    assert.dom('#test-panel-content').exists();
+  });
```

Comment:
Suggestion: add tests that aria-labelledby is always present, also that the correct role is set (depending on if the list is interactive or not)